### PR TITLE
Architecture

### PR DESCRIPTION
**Pull request by Jacob-Pang**
Added Appointment Diagram
**Note** : Sequence Diagram to be updated by @pngsebastian

### CODE REVIEW

File: docs/diagrams/AppointmentDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Class PatientMap PATIENT_MAP_COLOR
+Class DoctorMap DOCTOR_MAP_COLOR
+Class Appointment APPOINTMENT_COLOR
+Class Patient PATIENT_COLOR
+Class Doctor DOCTOR_COLOR
+Abstract Person PERSON_COLOR
+Class Timeslot TIMESLOT_COLOR
+
+Appointment --> PatientMap
+Appointment --> DoctorMap
```
Comment (icytornado) [inline docs/diagrams/AppointmentDiagram.puml:16]: should this be composition?

File: docs/diagrams/AppointmentDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+PatientMap --> Patient
+DoctorMap --> Doctor
+Appointment --> Timeslot
```
Comment (icytornado) [inline docs/diagrams/AppointmentDiagram.puml]: should this be composition?

File: docs/diagrams/AppointmentDiagram.puml (hunk 1 above)
Comment (Jacob-Pang) [reply docs/diagrams/AppointmentDiagram.puml:16]: No, PatientMap and DoctorMap are static global variables at runtime

File: docs/diagrams/AppointmentDiagram.puml (hunk 2 above)
Comment (Jacob-Pang) [reply docs/diagrams/AppointmentDiagram.puml]: Appointment, yes, will change accordingly.

**Review by onnwards — APPROVED** [review]

**Pull request merged**